### ISSUE TITLE
Focus loop crashes the browser

### ISSUE THREAD

**Issue by georgekaran** [ariakit/ariakit#2168]
### Current behavior

Using the keyboard navigation to open a Popover with no focusable elements causes the browser to crash.

### Steps to reproduce the bug

1. Open sandbox: https://codesandbox.io/s/clever-river-9rfl7w?file=/src/popover.tsx
2. Press the tab to focus on the button
3. Press enter to open the popover
4. Press the tab again
5. The browser freezes

### Expected behavior

The browser should not crash when no other element is available.

### Workaround

https://github.com/ariakit/ariakit/issues/2168#issuecomment-1366314645

### Possible solutions

This is probably related to the `preserveTabOrder` logic:

https://github.com/ariakit/ariakit/blob/7a575278ef438fdeb7f43859a67f14f6e31f7f9a/packages/ariakit/src/portal/portal.tsx#L177-L244

When there's no tabbable element, the focus traps likely move focus between themselves continuously, causing an infinite loop.

**Comment by diegohaz** [comment]
Thanks for reporting. It looks like it works if you use `requestAnimationFrame`:

```jsx
<button onClick={() => requestAnimationFrame(popover.show)}>
```

My first guess would be something is causing an infinite loop on <kbd>Enter</kbd> key up.

**Comment by georgekaran** [comment]
Thanks so much for the workaround. 

I've been digging a bit to find a definitive solution, and it seems the problem is in the hook `usePortal`, the focus trap is looping onto each other.

Adding the following branch to the `onFocus` callback solved the problem.
```tsx
{preserveTabOrder && portalNode && (
  <FocusTrap
    ref={beforeInsideRef}
    onFocus={(event) => {
      if (isFocusEventOutside(event, portalNode)) {
        const nextFocusableElement = getNextTabbable();

        // Prevent focus loop
        if (nextFocusableElement !== afterInsideRef.current) {
          queueFocus(getNextTabbable());
        }
      } else {
        queueFocus(beforeOutsideRef.current);
      }
    }}
  />
)}
```

Let me know what do you make of it, then I could open a PR.

**Comment by diegohaz** [comment]
That makes sense. We may also need to add the reversed logic to the `afterInside` element.

**Comment by georgekaran** [comment]
I just submitted a PR https://github.com/ariakit/ariakit/pull/2170 to fix this bug, let me know what do you think.
